sdk/core/azure-core/inc/azure/core/http/http.hpp

```diff
@@ -84,43 +84,24 @@ namespace Azure { namespace Core { namespace Http {
   /**
    * HTTP request method.
    */
-  enum class HttpMethod
-  {
-    Get, ///< GET
-    Head, ///< HEAD
-    Post, ///< POST
-    Put, ///< PUT
-    Delete, ///< DELETE
-    Patch, ///< PATCH
-  };
+  class HttpMethod {
+  public:
+    HttpMethod() = default;
+    explicit HttpMethod(std::string value) : m_value(std::move(value)) {}
+    bool operator==(const HttpMethod& other) const { return m_value == other.m_value; }
+    bool operator!=(const HttpMethod& other) const { return !(*this == other); }
```

Comment:
why is this defined in terms of the `==` operator?

Reply:
That's how it is in the guidelines and why wouldn't we want to define one in terms of the other? This way, we only have to implement one. Shouldn't `!=` always be `!(==)`?